crates/hir_ty/src/checker.rs

```diff
@@ -123,8 +138,9 @@ impl<'a> FunctionTypeChecker<'a> {
                 }
 
                 match callee {
-                    hir::Symbol::Local { expr, .. } => {
-                        self.check_expr(*expr);
+                    hir::Symbol::Local { binding, .. } => {
+                        let expr = binding.lookup(self.hir_file.db(self.db)).expr;
+                        self.check_expr(expr);
```

Comment:
**Performance Issue**: The modification to use binding.lookup introduces additional database queries within a loop, potentially leading to significant performance degradation due to repeated database access. <br> **Fix**: Cache the results of binding.lookup outside the loop to minimize database access and improve performance. <br> **Code Suggestion**: 
 ```
 +    let bindings = pod.all_functions(db).map(|(hir_file, function)| {
 +        (function, binding.lookup(hir_file.db(db)))
 +    }).collect::<HashMap<_, _>>();
 +    for (function, binding_expr) in bindings {
 +        self.check_expr(binding_expr.expr);
 +    }
 ```